packages/devtools_app/lib/src/screens/memory/framework/connected/memory_controller.dart

```diff
@@ -69,18 +69,32 @@ class MemoryFeatureControllers {
 class MemoryController extends DisposableController
     with AutoDisposeControllerMixin {
   MemoryController({
-    DiffPaneController? diffPaneController,
-    ProfilePaneController? profilePaneController,
+    @visibleForTesting DiffPaneController? diffPaneController,
+    @visibleForTesting ProfilePaneController? profilePaneController,
   }) {
     controllers = MemoryFeatureControllers(
       diffPaneController,
       profilePaneController,
     );
+    shareClassFilterBetweenProfileAndDiff();
   }
 
   /// Sub-controllers of memory controller.
   late final MemoryFeatureControllers controllers;
 
+  void shareClassFilterBetweenProfileAndDiff() {
+    controllers.diff.derived.applyFilter(controllers.profile.classFilter.value);
+
+    controllers.profile.classFilter.addListener(() {
+      controllers.diff.derived
+          .applyFilter(controllers.profile.classFilter.value);
+    });
+
+    controllers.diff.core.classFilter.addListener(() {
+      controllers.profile.setFilter(controllers.diff.core.classFilter.value);
```

Comment:
Just confirming, setting the filter in these callbacks doesn't trigger the other listener right?

Reply:
It does. Profile and diff controllers will do their own handling of the new filter.
But number of classes is not huge, so it is ok.